mopidy_spotify/backend.py

```diff
@@ -37,13 +39,18 @@ class SpotifyPlaybackProvider(backend.PlaybackProvider):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self._cache_location = Extension().get_cache_dir(self.backend._config)
+        self._data_location = Extension().get_data_dir(self.backend._config)
         self._config = self.backend._config["spotify"]
 
     def on_source_setup(self, source):
         for prop in ["username", "password", "bitrate"]:
             source.set_property(prop, str(self._config[prop]))
+        credentials_dir = os.path.join(self._data_location, "credentials-cache")
+        if not os.path.exists(credentials_dir):
+            os.makedirs(self._data_location, exist_ok=True)
+            os.mkdir(credentials_dir, 0o700)
+        source.set_property("cache-credentials", credentials_dir)
```

Comment:
We try and use Pathlib in new code e.g.

```suggestion
        credentials_dir = self._data_location / "credentials-cache"
        if not credentials_dir.exists():
            credentials_dir.mkdir(mode=0o700, exist_ok=True)
        source.set_property("cache-credentials", credentials_dir)
```

Reply:
We could also move this to `SpotifyPlaybackProvider.__init__()` (or even `SpotifyBackend.on_start()`) since it's really just a one-off concern.